File: ios/App/App/Helpers/Filesystem.swift
//
//  Filesystem.swift
//  App
//
//  Created by Julia Samol on 14.02.23.
//

import Foundation

enum Directory: String {
    case documents = "DOCUMENTS"
    case data = "DATA"
    case library = "LIBRARY"
    case cache = "CACHE"
    case external = "EXTERNAL"
    case externalStorage = "EXTERNAL_STORAGE"
}

extension FileManager {
    func getDirectory(from directory: Directory?) -> FileManager.SearchPathDirectory? {
        switch directory {
        case .library:
            return .libraryDirectory
        case .cache:
            return .cachesDirectory
        case .documents, .data, .external, .externalStorage:
            return .documentDirectory
        default:
            return nil
        }
    }
}

File: android/app/src/main/java/it/airgap/wallet/util/Filesystem.kt
package it.airgap.wallet.util

import android.content.Context
import android.os.Environment
import java.io.File

enum class Directory(val value: String) {
    Documents("DOCUMENTS"),
    Data("DATA"),
    Library("LIBRARY"),
    Cache("CACHE"),
    External("EXTERNAL"),
    ExternalStorage("EXTERNAL_STORAGE");

    companion object {
        fun fromString(value: String): Directory? = values().find { it.value == value }
    }
}

fun Context.getDirectory(directory: Directory?): File? =
    when (directory) {
        Directory.Documents -> Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS)
        Directory.Data, Directory.Library -> filesDir
        Directory.Cache -> cacheDir
        Directory.External -> getExternalFilesDir(null)
        Directory.ExternalStorage -> Environment.getExternalStorageDirectory()
        else -> null
    }
